mapie/tests/test_calibration.py

```diff
@@ -0,0 +1,390 @@
+    pred_ = mapie_cal.predict_proba(X_test)
+    top_label_ece_ = top_label_ece(y_test, pred_)
+    np.testing.assert_array_almost_equal(
+        results["y_score"], pred_  # type:ignore
+    )
+    np.testing.assert_allclose(  # type:ignore
+        results["top_label_ece"],
+        top_label_ece_,
+        rtol=1e-2
+    )
+
+
+def test_correct_results_binary() -> None:
+    """
+    Test that the y_score and top label score from the test dataset result
+    in the correct scores (in a binary setting).
+    """
+    X_binary, y_binary = make_classification(
+        n_samples=10,
+        n_classes=2,
+        n_informative=4,
+        random_state=random_state
+    )
+    mapie_cal = MapieCalibrator()
+    mapie_cal.fit(
+        X=X_binary,
+        y=y_binary,
+        random_state=random_state
+    )
+    pred_ = mapie_cal.predict_proba(X_binary)
+    top_label_ece_ = top_label_ece(y_binary, pred_)
+    ece = expected_calibration_error(y_binary, pred_)
+    np.testing.assert_array_almost_equal(
+        results_binary["y_score"], pred_  # type:ignore
+    )
+    np.testing.assert_allclose(  # type:ignore
+        results_binary["top_label_ece"],
+        top_label_ece_,
+        rtol=1e-2
+    )
+    np.testing.assert_allclose(  # type:ignore
+        results_binary["ece"],
+        ece,
+        rtol=1e-2
+    )
+
+
+def test_different_binary_y_combinations() -> None:
+    """
+    Test that despite the different maximum in y value, the
+    scores are always the same.
+    """
+    X_comb, y_comb = make_classification(
+        n_samples=20,
+        n_classes=3,
+        n_informative=4,
+        random_state=random_state
+    )
+    mapie_cal = MapieCalibrator()
+    mapie_cal.fit(X_comb, y_comb, random_state=random_state)
+    y_score = mapie_cal.predict_proba(X_comb)
+
+    y_comb1 = y_comb.copy()
+    y_comb1[np.where(y_comb1 == 2)[0]] = 3
+    mapie_cal1 = MapieCalibrator()
+    mapie_cal1.fit(X_comb, y_comb1, random_state=random_state)
+    y_score1 = mapie_cal1.predict_proba(X_comb)
+
+    y_comb2 = y_comb.copy()
+    y_comb2[np.where(y_comb2 == 2)[0]] = 40
+    mapie_cal2 = MapieCalibrator()
+    mapie_cal2.fit(X_comb, y_comb2, random_state=random_state)
+    y_score2 = mapie_cal2.predict_proba(X_comb)
+    np.testing.assert_array_almost_equal(y_score, y_score1)
+    np.testing.assert_array_almost_equal(y_score, y_score2)
+    assert top_label_ece(
+        y_comb, y_score, classes=mapie_cal.classes_
+    ) == top_label_ece(
+        y_comb1, y_score1, classes=mapie_cal1.classes_
+    )
+    assert top_label_ece(
+        y_comb, y_score, classes=mapie_cal.classes_
+    ) == top_label_ece(
+        y_comb2, y_score2, classes=mapie_cal2.classes_
+    )
+
+
+def test_results_with_constant_sample_weights() -> None:
+    """
+    Test predictions when sample weights are None
+    or constant with different values.
+
+    Note that it would seem that the calibration implementations
+    from sklearn would not pass these tests.
```

Comment:
Have you investigated?

Reply:
If you run the following code, you would see the error:


"
import numpy as np
from sklearn.calibration import _SigmoidCalibration
from sklearn.ensemble import RandomForestClassifier
from sklearn.base import clone
from sklearn.datasets import make_classification

random_state = 20

X, y = make_classification(
n_samples=20,
n_classes=2,
n_informative=4,
random_state=random_state
)

estimator = RandomForestClassifier(random_state=random_state)
estimator.fit(X, y)
probs_ = estimator.predict(X)

n_samples = len(X)

mapie_clf0 = _SigmoidCalibration()
mapie_clf1 = clone(mapie_clf0)
mapie_clf2 = clone(mapie_clf0)


mapie_clf0.fit(probs_, y, sample_weight=None)
y_pred0 = mapie_clf0.predict(probs_)

mapie_clf1.fit(probs_, y, sample_weight=np.ones(shape=n_samples))
y_pred1 = mapie_clf1.predict(probs_)

mapie_clf2.fit(probs_, y, sample_weight=np.ones(shape=n_samples) * 5)
y_pred2 = mapie_clf2.predict(probs_)


np.testing.assert_allclose(y_pred0, y_pred1)
np.testing.assert_allclose(y_pred0, y_pred2)
"